### PR TITLE
Updates MHV on VA.gov MyHealtheVet newsletter signup text

### PR DESCRIPTION
**Pull request by carlosfelix2**
## Summary
Updates the test for signing up for the newsletter and replaces the input field with just a link that opens in a new tab.

## Related issue(s)
https://app.zenhub.com/workspaces/mhv-on-vagov-landing-page-62619a987d74510018ecc546/issues/gh/department-of-veterans-affairs/va.gov-team/77515

## Testing done
- Unit test updated

## Screenshots
|         | Before | After |
| ------- | ------ | ----- |
| Mobile  |   staging va gov_my-health_postLogin=true     |   localhost_3001_my-health_ (7)   |
| Desktop |   staging va gov_my-health_postLogin=true (1)    |   localhost_3001_my-health_ (6)  |

## What areas of the site does it impact?
- MHV on VA.gov landing page

## Acceptance criteria
* Beneath the H2 for “subscribe to My HealtheVet updates”, a materially honest link opens the GovDelivery signup website in a new tab

### Quality Assurance & Testing

- [x] I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [x] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [x] Linting warnings have been addressed
- [x] Documentation has been updated ([link to documentation](#) \*if necessary)
- [x] Screenshot of the developed feature is added
- [x] [Accessibility testing](https://depo-platform-documentation.scrollhelp.site/developer-docs/wcag-2-1-success-criteria-and-foundational-testing) has been performed

### Error Handling

- [x] Browser console contains no warnings or errors.
- [x] Events are being sent to the appropriate logging solution
- [x] Feature/bug has a monitor built into Datadog or Grafana (if applicable)

### Authentication

- [x] Did you login to a local build and verify all authenticated routes work as expected with a test user

### :warning: Team Sites (only applies to modifications made to the VA.gov header) :warning:

- [ ] The vets-website header does not contain any web-components
- [ ] I used the [proxy-rewrite steps](https://github.com/department-of-veterans-affairs/vets-website/tree/main/src/applications/proxy-rewrite#local-dev) to test the injected header scenario
- [ ] I reached out in the `#sitewide-public-websites` Slack channel for questions

### CODE REVIEW

File: src/applications/mhv-landing-page/components/NewsletterSignup.jsx [hunk 1]
```diff
@@ -7,34 +7,24 @@ const NewsletterSignup = () => {
...
+      <div className="vads-l-row">
+        <a href={mhvNewsletterURL} rel="noopener noreferrer" target="_blank">
+          <i
+            aria-hidden="true"
+            role="img"
+            className="fas fa-chevron-circle-right fa-lg vads-u-color--link-default vads-u-margin-right--1"
+          />
+          Subscribe to the My HealtheVet newsletter on GovDelivery.com (opens in
+          new tab)
+        </a>
```
Comment (radavis) [inline src/applications/mhv-landing-page/components/NewsletterSignup.jsx:27]: You may want to use the Secondary Link - Action component, here, which includes the blue arrow icon.

https://design.va.gov/components/link/action#secondary

https://design.va.gov/storybook/iframe.html?id=components-action-link--reverse&viewMode=docs#secondary

**Review by radavis — DISMISSED** [review]
Left a suggestion, but not a blocker.

**Review by jzucadi — DISMISSED** [review]

File: src/applications/mhv-landing-page/components/NewsletterSignup.jsx (hunk 1 above)
Comment (carlosfelix2) [reply src/applications/mhv-landing-page/components/NewsletterSignup.jsx]: Definitely a must! Changed!

**Review by jzucadi — APPROVED** [review]

**Review by radavis — APPROVED** [review]

**Review by batemapf — APPROVED** [review]

**Comment by danbivins** [comment]
A11y QA is complete and no issues to report. Good to go. 

Tested with:
* Axe
* WAVE
* Browser Zoom to 400%
* JAWS + Edge
* VoiceOver + Safari

**Pull request merged**